contracts/carrot-app/src/handlers/query.rs

```diff
@@ -89,26 +85,33 @@ fn query_compound_status(deps: Deps, env: Env, app: &App) -> AppResult<CompoundS
 }
 
 fn query_position(deps: Deps) -> AppResult<PositionResponse> {
-    let position = POSITION.may_load(deps.storage)?;
-
-    Ok(PositionResponse { position })
+    let position_id = CarrotPosition::may_load(deps)?.map(|carrot_position| carrot_position.id);
+    Ok(PositionResponse { position_id })
 }
 
 fn query_config(deps: Deps) -> AppResult<Config> {
     Ok(CONFIG.load(deps.storage)?)
 }
 
 fn query_balance(deps: Deps, _app: &App) -> AppResult<AssetsBalanceResponse> {
-    let Some(position) = POSITION.may_load(deps.storage)? else {
+    let Some(carrot_position) = CarrotPosition::may_load(deps)? else {
         return Ok(AssetsBalanceResponse {
             balances: vec![],
             liquidity: "0".to_string(),
         });
     };
-    let pool = get_osmosis_position(deps, position.position_id)?;
 
-    let balances = try_proto_to_cosmwasm_coins(vec![pool.asset0.unwrap(), pool.asset1.unwrap()])?;
-    let liquidity = pool.position.unwrap().liquidity.replace('.', "");
+    let balances = try_proto_to_cosmwasm_coins(vec![
+        carrot_position.position.asset0.unwrap(),
+        carrot_position.position.asset1.unwrap(),
+    ])?;
+    // TODO: Use Decimal256 instead?
```

Comment:
Nah, FE needs to provide as integer as well. Could use `Uint256` but serialization will be the same (String).